chainer/optimizer_hooks/gradient_clipping.py

```diff
@@ -52,19 +82,21 @@ def __init__(self, threshold):
         self.threshold = threshold
 
     def __call__(self, opt):
-        sqnorm = _sum_sqnorm([p.grad for p in opt.target.params(False)])
-        device = backend.get_device_from_array(sqnorm)
+        sqnorm, device = _sum_sqnorm_grads(list(opt.target.params(False)))
+        if device is None:
+            # Assign a dummy device for using_device.
+            device = backend.CpuDevice()
+
         with chainer.using_device(device):
             norm = device.xp.sqrt(sqnorm)
             rate = self.threshold / norm
             # When no clipping is needed, skip the clipping on CPU and
             # multiply 1.0 on the device otherwise.
-            if device.xp is numpy:
-                if rate >= 1:
-                    return
-            else:
-                rate = rate.clip(None, 1)
+            if device.xp is numpy and rate >= 1:
```

Comment:
Should we not skip the scaling for non-numpy (when rate >= 1)?

Reply:
Fixed.

Reply:
Thanks, could you also update the comment associated with this logic?

Reply:
Hm, did this fix get lost?

Reply:
It seems both of us were confused on this fix.
I updated the comment to make it more clear (hopefully).